### ISSUE TITLE
Volume Name of Dokany mounted Filesystem longer than 32 characters

### ISSUE THREAD

**Issue by infeo** [dokan-dev/dokany#939]
# Environment
 * Windows version: 10 Pro 2004
 * Processor architecture: x64
 * Dokany version: Current Snapshot
 * Library type (Dokany/FUSE): Dokany
 
# Check List
  - [x] I checked my issue [doesn't exist yet](https://github.com/dokan-dev/dokany/issues?utf8=%E2%9C%93&q=is%3Aissue)
  - [x] My issue is valid with mirror default sample and not specific to my user-mode driver implementation
  - [x] I can always reproduce the issue with the provided description below.
  - [x] I have updated Dokany to the [latest version](https://github.com/dokan-dev/dokany/releases) and have reboot my computer after.
  - [x] I tested one of the last [snapshot](https://github.com/dokan-dev/dokany/wiki/Build#user-snapshot) from appveyor CI

# Description
According to the [Windows documenation](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setvolumelabelw) a volume label consists of at most 32 characters. In contrast, in Dokany the volume name (and hence its label)  has a length of MAX_PATH characters:
https://github.com/dokan-dev/dokany/blob/2ecdf599c773ee82512b1568ff238cd410385611/dokan/volume.c#L57-L61

My question is now, who is correct? It works with more than 32 characters apparently, but may be the cause for incompatabilities. Additionally, with the alternative [WinFSP](https://github.com/billziss-gh/winfsp) mounting fails if a name longer than 32 characters is used.

**Comment by Liryna** [comment]
Hi @infeo ,

Well from [FileFsVolumeInformation](https://github.com/dokan-dev/dokany/blob/master/sys/volume.c#L321-L341) `If the volume label is greater than 32 characters, return the first 32 characters of the label and STATUS_SUCCESS.` it sounds like it can be longer but only the first 32 characters will be taken.
Even fastfat truncate the volume name
https://github.com/microsoft/Windows-driver-samples/blob/16022c593787aef040b8a37ef697316e07f1322e/filesys/fastfat/volinfo.c#L642-L652
but set the full name length.

I would suggest here to:
* Let the current code use `MAX_PATH`
* Set the full VolumeName length in the Kernel https://github.com/dokan-dev/dokany/blob/master/sys/volume.c#L336
* Recommend the usage for 32 max length Volume name in GetVolumeInformation dokan API

We could also question the fsName length but nothing in the documentation limit the size [FileFsAttributeInformation](
https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/ebc7e6e5-4650-4e54-b17c-cf60f6fbeeaa).
But for this one, I would always recommend to use `NTFS` or `FAT` depending on the need.

**Comment by infeo** [comment]
> it sounds like it can be longer but only the first 32 characters will be taken.

Interersting. Well, the Windows Explorer definitely does not truncate the volume name^^:
![label_longer_than32](https://user-images.githubusercontent.com/9036915/98105737-a5c5d480-1e98-11eb-9f32-b534ef659a21.png)

I think thour suggestions are a good solution by not breaking anything, but informing the user to be wary about this fact. Do you want me to make a PR?

**Comment by Liryna** [comment]
@infeo A PR would be very welcomed as always 😉! Thanks in advance

**Comment by infeo** [comment]
To create the PR, I had another look at the documentation. In the doc for [GetVolumeInfomation](https://docs.microsoft.com/de-de/windows/win32/api/fileapi/nf-fileapi-getvolumeinformationw), it states explicitly, that the volume label is only restricted by the parameter `nVolumeNameSize` and nothing else.

I think, i had a wrong understanding of the volume size restriction: It is _filesystem specific_ by API design. So the one implementing the filesystem can restrict it to less than MAX_PATH+1, but it does not have to be. And the link in my first post just states the restriction for two specific file systems, NTFS and FAT, it does not hold in general.

The bottom line: imho there is no PR needed.

**Comment by Liryna** [comment]
@infeo That's also how I understand it. I am fine to let it as it is. If you still feel like you wouldn't had to do all this research with some doc added to dokan, do not hesitate.

Otherwise, always glad to hear you found your answers!

**Comment by Liryna** [comment]
@infeo FYI I added a comment to explain the limitation here https://github.com/dokan-dev/dokany/commit/f8f65f725ea7bd145642586c330702b7b4c81b38
